mtwitter.js

```diff
@@ -0,0 +1,254 @@
+                user: user
+            };
+            resolve(user);
+        });
+    });
+
+}
+
+/*
+  Add a user's timeline to the iterator. Tweets will be automatically send to
+  the given room.
+*/
+mtwitter.prototype.enqueue_timeline = function(userid, localroom, remoteroom) {
+    var obj = {
+        "user_id": userid,
+        "local": localroom,
+        "remote": remoteroom
+    };
+    this.timeline_list.push(obj);
+    this.timeline_queue.push(obj);
+}
+
+/*
+  This function will fill the content structure for a new matrix message
+  for a given tweet.
+*/
+mtwitter.prototype.construct_message = function(tweet, type) {
+    return {
+        "body": tweet.text,
+        "created_at": tweet.created_at,
+        "likes": tweet.favorite_count,
+        "reblogs": tweet.retweet_count,
+        "tweet_id": tweet.id_str,
+        "tags": tweet.entities.hashtags,
+        "msgtype": type
+    }
+}
+
+/*
+  Process a given tweet (including resolving any parent parent tweets), and
+  submit it to the appropriate room.
+*/
+mtwitter.prototype.process_tweet = function(bridge, roomid, tweet, treeN) {
+    //console.log(tweet);
+    var muser = "@twitter_" + tweet.user.id_str + ":" + bridge.opts.domain;
+    var intent = bridge.getIntent(muser);
+    treeN--;
+    if (treeN < 0) {
+        console.log("Bailing because we have gone too far deep.")
+        return;
+    }
+
+    var type = "m.text";
+    if (tweet.in_reply_to_status_id_str != null) {
+        type = "m.notice"; // A nicer way to show previous tweets
+    }
+    var tweet_content = this.construct_message(tweet, type);
+
+    if (tweet.in_reply_to_status_id_str != null) {
+        this.app_twitter.get('statuses/show/' + tweet.in_reply_to_status_id_str, {}, (error, newtweet, response) => {
+            if (!error) {
+                this.process_tweet(bridge, roomid, newtweet, treeN);
+                return;
+            }
+            console.error(error);
+        });
+        setTimeout(() => {
+            intent.sendEvent(roomid, "m.room.message", tweet_content)
+        }, treeN * 250); //Make sure not to send them too quickly.
+    }
+}
+
+/*
+  Internal function to process the timeline queue.
+*/
+mtwitter.prototype._process_timeline = function(self) {
+    if (self.timeline_queue.length < 1) {
+        return;
+    }
+    
+    var tline = self.timeline_queue[0];
+    self.timeline_queue = self.timeline_queue.slice(1);
+    var id = tline.remote.getId().substr(9);
+
+    var req = {
+        user_id: id,
+        count: 3
+    };
+    var since = tline.remote.get("twitter_since");
+    if (since != undefined) {
+        req.since_id = since;
+    }
+
+    self.app_twitter.get('statuses/user_timeline', req, (error, feed, response) => {
+        if (!error) {
+            if (feed.length > 0) {
+                tline.remote.set("twitter_since", feed[0].id_str, 3);
+                feed = feed.reverse();
+                for (var item in feed) {
```

Comment:
Misleading variable name which makes me think you aren't sure what the `in` operator is doing, even though you use it correctly on the next line. There are *very very few* valid reasons to use any of the `in` operators (including `for ... in`). It is a horrible operator, and I strongly discourage its use as there are clearer alternatives in ES5 and above.

Have some documentation which details why they are horrible operators:
 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in
 - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in

What do you do instead then? Simple. `forEach()`:

```javascript
feed.forEach((item) => {
  self.process_tweet(self._bridge, tline.local.roomId, item, 3);
});
```